libbeat/publisher/queue/memqueue/eventloop.go

```diff
@@ -121,13 +121,20 @@ func (l *directEventLoop) run() {
 		case req := <-getChan: // consumer asking for next batch
 			l.handleGetRequest(&req)
 
+		case req := <-l.broker.metricChan: // broker asking for queue metrics
+			l.handleMetricsRequest(&req)
+
 		case schedACKs <- l.pendingACKs:
 			// on send complete list of pending batches has been forwarded -> clear list
 			l.pendingACKs = chanList{}
 		}
 	}
 }
 
+func (l *directEventLoop) handleMetricsRequest(req *metricsRequest) {
+	req.responseChan <- memQueueMetrics{currentQueueSize: len(l.buf.entries)}
```

Comment:
`len(l.buf.entries)` returns the total capacity of the queue, not its current number of events. The current event count (including events that have been read by consumers but not yet acked/removed from the queue) would be `l.buf.regA.size + l.buf.regB.size` (it probably makes sense to make this a helper function on `ringBuffer` since it's not obvious).